deeprank2/trainer.py

```diff
@@ -210,30 +212,33 @@ def _init_datasets(self,  # pylint: disable=too-many-arguments
             else:
                 _log.warning("Validation dataset was provided to Trainer; val_size parameter is ignored.")
 
-        # Copy settings from the dataset that we will use.
-        if self.dataset_train is not None:
-            self._init_from_dataset(self.dataset_train)
-        else:
-            self._init_from_dataset(self.dataset_test)
-
     def _init_from_dataset(self, dataset: Union[GraphDataset, GridDataset]):
 
         if isinstance(dataset, GraphDataset):
             self.clustering_method = dataset.clustering_method
             self.node_features = dataset.node_features
             self.edge_features = dataset.edge_features
             self.features = None
+            self.features_transform = dataset.features_transform
+            self.means = dataset.means
+            self.devs = dataset.devs
 
         elif isinstance(dataset, GridDataset):
             self.clustering_method = None
             self.node_features = None
             self.edge_features = None
             self.features = dataset.features
+            self.features_transform = None
+            self.means = None
+            self.devs = None
         else:
             raise TypeError(type(dataset))
```

Comment:
Would be nice to return a better error message here. Maybe something like (although not perfect, because to the user it's unclear what `dataset` is in this case).:

```suggestion
        else:
    raise TypeError(
        f"Incorrect `dataset` type provided: {type(dataset)}. Please provide a `GridDataset` or `GraphDataset` object instead."
    )
```